cmd/pullrequest-init/github.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"golang.org/x/oauth2"
+
+	"github.com/google/go-github/github"
+	"go.uber.org/zap"
+)
+
+// GitHubHandler handles interactions with the GitHub API.
+type GitHubHandler struct {
+	*github.Client
+
+	owner, repo string
+	prNum       int
+
+	Logger *zap.SugaredLogger
+}
+
+// NewGitHubHandler initializes a new handler for interacting with GitHub
+// resources.
+func NewGitHubHandler(ctx context.Context, logger *zap.SugaredLogger, rawURL string) (*GitHubHandler, error) {
+	token := strings.TrimSpace(os.Getenv("GITHUBOAUTHTOKEN"))
+	var hc *http.Client
+	if token != "" {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		)
+		hc = oauth2.NewClient(ctx, ts)
+	}
+
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+	split := strings.Split(u.Path, "/")
+	if len(split) != 5 {
+		return nil, fmt.Errorf("could not determine PR from URL: %v", rawURL)
+	}
+	owner, repo, pr := split[1], split[2], split[4]
```

Comment:
hm could we maybe have a function that does this parsing and has its own unit tests?